src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -329,6 +329,11 @@ internal virtual bool IsExplicitInterfaceImplementation
 
         protected bool IsValidReadOnlyTarget => !IsStatic && ContainingType.IsStructType() && MethodKind != MethodKind.Constructor;
 
+        /// <summary>
+        /// Indicates whether synthesized methods derived from this method should inherit the attributes of this method, its parameters, return type, and type parameters.
+        /// </summary>
+        internal virtual bool SynthesizedMethodsInheritAttributes => false;
```

Comment:
>internal virtual bool SynthesizedMethodsInheritAttributes => false; [](start = 8, length = 67)

I find this approach very fragile and error prone. There could be multiple synthesized methods of different nature associated with the same "base" method. The property whether attributes should be inherited belongs to the specific synthesized method instead. Please make the change.